fix: handle livePhotos using originFileWithSubType

shenlong-tanwen opened the pull request:
####  Changes made:

- Live photo video assets are fetched using `originFileWithSubtype`
- `isLocallyAvailable` now checks for the existence of the `originFile`

shenlong-tanwen committed:
fix: handle livePhotos using originFileWithSubType

shenlong-tanwen committed:
remove livePhoto asset cache

shenlong-tanwen committed:
fetch live photo video name from entity

alextran1502 commented:
The latest changes somehow caused the MOV portion to be uploaded as an image and the server wasn't able to generate thumbnails or perform other processes on it

shenlong-tanwen committed:
fix: video file not detected

shenlong-tanwen commented:
> The latest changes somehow caused the MOV portion to be uploaded as an image and the server wasn't able to generate thumbnails or perform other processes on it

Fixed. Thank you! The `titleAsyncWithSubType` was not returning the proper video name as I thought. Previously, we used to fetch the video file name from the base path of the video file. This is resulting in huge file name with a UUID. I've updated the handling to use the same name as the image file with the extension of the video file. Should we leave it this way or go back to using the base path of the video file instead?

**The image file name:** "IMG_2211.HEIC"

**The video file name:** 
The X are random numbers. Probably some meta related to where it is stored along with it's timestamp.
**Before (base path):** "3B5DB4FE-9B59-4C08-B2F4-4D5C83FC50BB_XX_XXX_XXXXXXXXXX.XXXXXX_o_IMG_2211.MOV"
**After (image name + video ext):** "IMG_2211.MOV"

Also, I've verified upload and download of a live photo available locally. Can you test if the same works for a live photo directly from iCloud? Need to confirm that the following actually returns the video file as well
https://github.com/immich-app/immich/blob/70c4c38c891a5cdc18a24b5ca33312567f29e32e/mobile/lib/modules/backup/services/backup.service.dart#L275-L277

shenlong-tanwen committed:
chore: pull main

alextran1502 committed:
fix: set correct header
